base/compiler/ssair/ir.jl

```diff
@@ -1363,7 +1363,7 @@ function process_node!(compact::IncrementalCompact, result_idx::Int, inst::Instr
             end
         elseif !isa(pi_val, AnySSAValue) && !isa(pi_val, GlobalRef)
             valtyp = isa(pi_val, QuoteNode) ? typeof(pi_val.value) : typeof(pi_val)
-            if valtyp === stmt.typ
+            if valtyp === widenconst(stmt.typ)
```

Comment:
Maybe `widenconst` call can be expensive here? We may want to make it `stmt.typ isa Type ? stmt.typ : widenconst(stmt.typ)`?

Reply:
I don't think this is sound - the widening is in the wrong direction. I think what you want is
```suggestion
            if isa(stmt.typ, Const) ? pi_val === stmt.typ.val : valtyp ===stmt.typ
```

Reply:
True, we should check if `pi_val` is identical to the const type.